crates/wysiwyg/src/composer_model/format.rs

```diff
@@ -225,9 +226,16 @@ where
         locations: Vec<DomLocation>,
         format: &InlineFormatType,
     ) {
+        let mut moved_handles = HashMap::<DomHandle, DomHandle>::new();
         // Go through the locations in reverse order to prevent Dom modification issues
         for loc in locations.iter().rev() {
-            if Self::needs_format(&self.state.dom, loc, format) {
+            let mut loc = loc.clone();
+            let moved_parent_handle = moved_handles.get(&loc.node_handle);
+            if let Some(new_handle) = moved_parent_handle {
+                // Careful here, the location's position is no longer valid
+                loc = loc.with_new_handle(new_handle.clone());
```

Comment:
I tried to avoid modifying the location here and using DomHandle instead, but `split_text_node_by_offsets` relies on several DomLocation values and functions.